presto-product-tests-launcher/src/main/resources/docker/presto-product-tests/run-presto.sh

```diff
@@ -21,8 +21,10 @@ fi
 export JAVA_HOME="/usr/lib/jvm/zulu-11"
 export PATH="${JAVA_HOME}/bin:${PATH}"
 
+# When Presto OOM exception occurs, mark Presto as tainted and wait for health check
 exec /docker/presto-server/bin/launcher \
   -Dnode.id="${HOSTNAME}" \
   --etc-dir="/docker/presto-product-tests/conf/presto/etc" \
   --data-dir=/var/presto \
+  -J "-XX:OnOutOfMemoryError=echo Out of memory occured in pid %p;touch /tmp/presto_oom;sleep 30d" \
```

Comment:
This is very ha^H^H clever solution.

Could you please add a comment why we need the `30d` sleep here. It is very non obvious.

Also did you try to pursue the option where we change test-containers so they allow copying out stuff from finished container?

Reply:
Yes, I'm doing that in parallel but this still won't be easy as I need to exec find in a container to list files (requires running container) and copy files out of the container (doesn't require running container but testcontainers prohibits this)

This solution is an easy attempt to address that without any deep changes to copying logic

Reply:
As I understand the heap dump will still be made via
```
-XX:+HeapDumpOnOutOfMemoryError
```

Reply:
yes, exactly

I'm just adding additional java opt using launcher's mechanism (I prefer it here instead of jvm.config as it's more obvious what's the purpose of it)

Reply:
Alternatively we could mount the `/var/presto` directory into some temporary directory on the host created during `presto-master` startup, and get the heap dump from the host.

Reply:
I think we tried that in the past and builds were hanging weirdly. Maybe the reason was unrelted.